src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -26,26 +27,37 @@ public class StoreDatasetValidator : IStoreDatasetValidator
     private readonly bool _enableFullDicomItemValidation;
     private readonly IElementMinimumValidator _minimumValidator;
     private readonly IQueryTagService _queryTagService;
+    private readonly IValidationResultBuilderFactory _validationResultBuilderFactory;
+
 
-    public StoreDatasetValidator(IOptions<FeatureConfiguration> featureConfiguration, IElementMinimumValidator minimumValidator, IQueryTagService queryTagService)
+    public StoreDatasetValidator(
+        IOptions<FeatureConfiguration> featureConfiguration,
+        IElementMinimumValidator minimumValidator,
+        IQueryTagService queryTagService,
+        IValidationResultBuilderFactory validationResultBuilderFactory)
```

Comment:
I'm not sure we need DI for this - since this validation result is already specific to this usage I think we can just return the results as a data transfer object.